app/jobs/request_job.rb

```diff
@@ -0,0 +1,39 @@
+require 'faraday'
+
+class RequestJob < ApplicationJob
+  queue_as :default
+  retry_on Faraday::TimeoutError
+
+  rescue_from(Faraday::ClientError) do |e|
```

Comment:
How would you handle these kind of errors in production ready code?

Reply:
Async HTTP requests are often used to not block event queues by omitting response waiting time for the user. If the response was crucial(e.g user registration form) we had to use synchronous code to wait for the response and ask for the corrections from the user.
So, we have an async request and the error handling depends on the situation of the software. In my opinion, one of the common solutions would be informing users about a problem after failure. That could be sending a failure email (as yahoo or other email services does when a message has not been sent) or outputting it somewhere in UI on runtime. I assumed outputting it to the console would be enough to demonstrate the purpose.

Reply:
> How would you handle these kind of errors in production ready code?

Async HTTP requests are often used to not block event queues by omitting response waiting time for the user. If the response was crucial(e.g user registration form) we had to use synchronous code to wait for the response and ask for the corrections from the user.
So, we have an async request and the error handling depends on the situation of the software. In my opinion, one of the common solutions would be informing users about a problem after failure. That could be sending a failure email (as yahoo or other email services does when a message has not been sent) or outputting it somewhere in UI on runtime. I assumed outputting it to the console would be enough to demonstrate the purpose.

Reply:
Are you familiar with any job queue backends?

Reply:
Got the point now! To be honest I didn't know about the Applications Jobs because I have never dig deeper into Rails beyond basics. Thanks for giving useful tips.